neqo-http3-server/src/main.rs

```diff
@@ -385,7 +397,9 @@ fn main() -> Result<(), io::Error> {
                         if let Some((ref mut server, svr_timeout)) =
                             servers.get_mut(&socket.local_addr().unwrap())
                         {
-                            process(&mut **server, svr_timeout, inx, None, &mut timer, socket);
+                            if process(&mut **server, svr_timeout, inx, None, &mut timer, socket) {
+                                active_servers.insert(inx);
+                            }
```

Comment:
Why is this block not the same as the block you use when datagrams arrive or when a server is active?